sw/host/cryptotest/testvectors/parsers/wycheproof_ecdh_parser.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import jsonschema
+import logging
+import sys
+
+from Crypto.PublicKey import ECC
+
+# Mapping from the curve names used by Wycheproof to those used by cryptotest
+EC_NAME_MAPPING = {
+    "secp256r1": "p256",
+    "secp384r1": "p384",
+}
+
+
+def parse_test_vectors(raw_data):
+    test_groups = raw_data["testGroups"]
+    test_vectors = list()
+    for group in test_groups:
+        # Parse tests within the group
+        for test in group["tests"]:
+            logging.debug(f"Parsing tcId {test['tcId']}")
+            # Reject tests with invalid ASN
+            if "InvalidAsn" in test["flags"]:
+                logging.info(
+                    f"Skipped tcId {test['tcId']}: Tagged as Invalid ASN."
+                )
+                continue
+            # Wycheproof flags one (valid) test using a compressed key
+            # encoding as "acceptable".  Because all other
+            # "acceptable" tests should be rejected, we throw this one
+            # out.
```

Comment:
I'm not sure I understand this comment -- is the parser unable to handle compressed keys? Why is only the "acceptable" test-case being thrown out?

Reply:
Unfortunately, when I changed to using `ECC.import_key()` to parse the DER-encoded public keys, one test case for each of the curves uses the compressed ECC point encoding, which `import_key()` has no problem with. For some reason wycheproof flagged this test as `acceptable`, not `valid`, and because we are (rightfully so) rejecting all other `acceptable` tests, this test had to be thrown out. Note that the explicit check for the `acceptable` result was not necessary for functional correctness, but I tried to be as specific as possible in case other actually invalid tests using the `CompressedPoint` encoding are added later.

Reply:
It's the latter. There's nothing "wrong" with the test (the compressed encoding gives the same (x, y) public key as the test right above it). So cryptolib accepts it but wycheproof marks it as `acceptable`.

Reply:
Chatted offline about this:

It's probably preferable in this case to still run the test vector if compressed key turns into a valid public key. The comment can then also be a bit clearer. In the case of this particular test vector, where the public key is same as that of the previous test, it's redundant, but in case a test vector is added later that does not include a "non-compressed" counterpart, we would like to not exclude it from the testing.